packages/headless/src/controllers/facets/range-facet/headless-range-facet.ts

```diff
@@ -65,6 +71,15 @@ export function buildRangeFacet<
       return this.state.sortCriterion === criterion;
     },
 
+    updateRangeAlgorithm(rangeAlgorithm: RangeFacetRangeAlgorithm) {
```

Comment:
You would want users to define the range algorithm during the configuration of the facet, it should be an option for the range facets.

Switching the algorithm afterwards, once a request has been performed, would cause a discrepancy between the ranges you already received.

I would then remove the methods and opt towards an option instead, like the `generateAutomaticRanges` or `sortCriteria` option.

Reply:
I see, you only set it at the creation of the facet, you are not allowed to change it afterwards.

Reply:
Yep, it does not need to be dynamic,